File: cashier/src/main/java/com/secwager/cashier/di/CashierComponent.java
package com.secwager.cashier.di;

import com.secwager.cashier.CashierServiceImpl;
import dagger.Component;
import javax.inject.Singleton;

@Singleton
@Component(modules = CashierModule.class)
public interface CashierComponent {

  CashierServiceImpl buildCashierService();

}

File: cashier/src/main/kotlin/di/CashierComponent.kt
package com.secwager.cashier.di

import com.secwager.cashier.CashierServiceImpl
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [CashierModule::class])
interface CashierComponent {
    fun buildCashierService(): CashierServiceImpl?
}